feat: include reference to issue closed with a PR

maelle opened the pull request:
Fix #361

maelle commented on R/update-news.R:
```diff
@@ -348,6 +394,30 @@ check_gh_pat <- function() {
+  if (length(missing_scopes) > 0) {
+        rlang::warn(
+      message = c(
+        x = sprintf("Missing scopes for GitHub GraphQL API (used for finding issues linked to PR): %s", toString(missing_scopes)),
```
still not sure whether we should warn here

maelle replied:
(as I'm unsure which missing scope is really an issue, it's not documented, it seems over the top to try them all)

maelle replied:
@krlmlr opinion?

krlmlr replied:
Is this the right place? We give a hard error elsewhere if scopes are missing. (I typically set `GITHUB_PAT` manually in every session I need it.)

maelle replied:
The check that leads to an error is done in the same function just before this. There is also `check_gh_scopes()` that is used in code that really needs the repo scope for opening PRs.

The thing is, I'm not sure not having the right scopes warrants an error especially as the fledge user might not even use the PR message feature.

`check_gh_scopes()` and `check_gh_pat()` could be merged and have an argument that'd define what scope is checked and whether it's a show stopper, however before doing that I'd like to confirm it's ok to **not** error if there are not the V4 scopes (which only means we can't "include reference to issue closed with a PR").

krlmlr replied:
Failing hard is okay for now. Once we have a way to add this information later on, we can turn this into a message (and/or an entry in `NEWS.md`) with corresponding instructions.

krlmlr commented on R/update-news.R:
```diff
@@ -320,9 +357,16 @@ harvest_pr_data <- function(message) {
     }
   }
 
+  issue_numbers <- if ("closingIssuesReferences" %in% names(issue_info$data$repository$pullRequest)) {
```
Do we even need the `if {}` here?

krlmlr reviewed (commented):
Thanks, I'll also check how this works in practice.

krlmlr commented on R/update-news.R:
```diff
@@ -320,9 +357,19 @@ harvest_pr_data <- function(message) {
     }
   }
 
+  issue_numbers <- purrr::map(
+    issue_info$data$repository$pullRequest$closingIssuesReferences$edges,
+    ~ .x$node$number
+  )
+  if (length(issue_numbers) == 0) {
+    issue_numbers <- NULL
+  }
```
Perhaps:

```suggestion
  issue_numbers <- purrr::map_int(
    issue_info$data$repository$pullRequest$closingIssuesReferences$edges,
    ~ as.integer(.x$node$number)
  )
```

krlmlr commented on tests/testthat/_snaps/update-news.md:
```diff
@@ -56,14 +71,17 @@
+      # A tibble: 1 x 4
+        description                                                type  break~1 scope
+        <chr>                                                      <chr> <lgl>   <lgl>
+      1 PLACEHOLDER https://github.com/cynkra/fledge/pull/332 (#3~ Unca~ FALSE   NA   
```
Why are we moving away from JSON here?

maelle replied:
Because if we used JSON here we would not snapshot the message "Could not get title for PR #332 (no internet connection)". 

Should I tweak `expect_snapshot_tibble()` further so it'd snapshot everything?

krlmlr replied:
If that message is an error, we don't need to care about this just yet?

maelle replied:
Ok

krlmlr commented on tests/testthat/_snaps/update-news.md:
```diff
@@ -78,9 +96,10 @@
       <simpleError in gh::gh(glue("GET /repos/{slug}/pulls/{pr_number}")): bla>
     Message
       ! Could not get title for PR #332
+      No encoding supplied: defaulting to UTF-8.
```
This looks like we could fix it?

maelle replied:
Not really https://github.com/nealrichardson/httptest/issues/75

For another test I can use suppressMessages() but in this one we are recording our own message as well.

krlmlr commented on R/update-news.R:
```diff
@@ -186,6 +186,12 @@ add_squash_info <- function(description) {
 parse_merge_commit <- function(message) {
   pr_data <- harvest_pr_data(message)
   pr_number <- pr_data$pr_number
```
Do we still need this?

```suggestion
```

krlmlr reviewed (commented):
Can you please merge/rebase main to remove the noise in the snapshots?

maelle commented:
@krlmlr there are two pending conversations above regarding the tests.

krlmlr commented on R/auto.R:
```diff
@@ -392,36 +392,17 @@ merge_branch <- function(other_branch) {
 
 check_post_release <- function() {
-  cli_alert("Checking scope of {.var GITHUB_PAT} environment variable.")
+  cli_alert("Checking presence and scope scope of {.var GITHUB_PAT}.")
```
I don't understand "everyone".

```suggestion
  cli_alert("Checking presence and scope of {.var GITHUB_PAT}.")
```

krlmlr reviewed (commented):
I still don't understand why we switched from `expect_snapshot_tibble()` to `expect_snapshot()` .

maelle commented:
> I still don't understand why we switched from expect_snapshot_tibble() to expect_snapshot() .

Well now that I read you're ok with ignoring the message, I switched back. :smile_cat:

krlmlr commented:
Thanks!

Eventually we should make all kinds of failures (no internet, can't scrape) a "hard" error, with the possibility to opt out.

krlmlr commented:
Warnings are too easy to miss.
